packages/beacon-node/src/eth1/eth1DepositDataTracker.ts

```diff
@@ -112,7 +112,7 @@ export class Eth1DepositDataTracker {
   /**
    * Return eth1Data and deposits ready for block production for a given state
    */
-  async getEth1DataAndDeposits(state: BeaconStateAllForks): Promise<Eth1DataAndDeposits> {
+  async getEth1DataAndDeposits(state: CachedBeaconStateAllForks): Promise<Eth1DataAndDeposits> {
```

Comment:
`if (state.eth1_deposit_index >= state.deposit_receipts_start_index)` return `eth1Data` from the state and empty `deposits`